apps/aecore/src/aec_db.erl

```diff
@@ -588,8 +599,15 @@ find_tx_location(STxHash) ->
            [] ->
                case mnesia:read(aec_tx_pool, STxHash) of
                    [] -> case mnesia:read(aec_signed_tx, STxHash) of
-                             [] -> not_found;
-                             [_] -> none
+                             [] ->
+                                 case mnesia:read(aec_tx_gc, STxHash) of
+                                     []  -> not_found;
+                                     [_] -> none
```

Comment:
This leads to the doubt of how can actor (human or automated) calling an API for pushing to mempool distinguish case tx inserted and case tx not pushed i.e. [this clause](https://github.com/aeternity/aeternity/blob/e9bd96779c585eab037727b008babce242dab2da/apps/aecore/src/aec_tx_pool.erl#L172). Though this is not introduced by this change so ok.

This though leads to doubt of whether there is any actor relying on pool push returning ok, and retrieval of tx succeeding.

Reply:
> This though leads to doubt of whether there is any actor relying on pool push returning ok, and retrieval of tx succeeding.

Still applicable.

Reply:
I did not find said code path in the code...

Reply:
> I did not find said code path in the code...

If does not need necessarily to be part of our code. It can be any user of the user API.

Also, it may manifest itself in client (either in node or out of node) using push for ensuring that certain tx is present in mempool - the semantics of ok there is changed.

Reply:
> Also, it may manifest itself in client (either in node or out of node) using push for ensuring that certain tx is present in mempool - the semantics of ok there is changed.

This sentence is incorrect. Push returning ok in master already does not guarantee tx in mempool if garbage collected from mempool.